src/wallet/rpcwallet.cpp

```diff
@@ -239,6 +245,56 @@ UniValue getaddressinfo(const UniValue& params, bool fHelp)
     return ret;
 }
 
+void parseWIFKey(std::string strKey, CWallet* pwallet, CPubKey& master_pub_key) {
```

Comment:
nit: think it would be more readable to return the public key, i.e.
```cpp
CPubKey parseWIFKey(std::string strKey, CWallet* pwallet)
{
    .
    .
    .
    return pwallet->GetScriptPubKeyMan()->DeriveNewSeed(key);
}
```

This way, in `sethdseed` you could use a more compact assignment
```cpp
CPubKey master_pub_key = params[1].isNull() ?
                         spk_man->GenerateNewSeed() :
                         parseWIFKey(params[1].get_str(), pwallet);
```